compiler/dialects/acc.py

```diff
@@ -0,0 +1,163 @@
+from collections.abc import Iterable, Sequence
+
+from xdsl.dialects.builtin import ArrayAttr, StringAttr
+from xdsl.ir import (
+    Attribute,
+    Dialect,
+    Operation,
+    ParametrizedAttribute,
+    SSAValue,
+    TypeAttribute,
+)
+from xdsl.irdl import (
+    AttrSizedOperandSegments,
+    IRDLOperation,
+    ParameterDef,
+    VerifyException,
+    irdl_attr_definition,
+    irdl_op_definition,
+    operand_def,
+    opt_operand_def,
+    prop_def,
+    result_def,
+    var_operand_def,
+)
+
+
+@irdl_attr_definition
+class TokenType(ParametrizedAttribute, TypeAttribute):
+    """
+    Async token type for launched accelerator requests.
+    """
+
+    name = "acc.token"
```

Comment:
The `TokenType` is empty: let me just check if I get the working of this correctly: The sole purpose of a token is to easily know which `acc.wait` and `acc.launch` are linked together, which is useful in some passes? This won't actually be something that is lowered, as the `acc.await` will be based on the accelerator id, not the token.

Reply:
Yes, pretty much. The token is used so we can trace which acccelerator to wait on. E.g. the launch launches an accelerator, and this way the await will always match the launch accelerator.

This also allows us to maybe infer the kernel runtimes, so we can push setup calls up before the previous await, saving more cycles!